sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -263,10 +261,9 @@ private static BiFunction<HttpClientResponse, Connection, Publisher<HttpResponse
              */
             if (eagerlyReadResponse) {
                 // Set up the body flux and dispose the connection once it has been received.
-                return FluxUtil.collectBytesFromNetworkResponse(
-                        reactorNettyConnection.inbound().receive().asByteBuffer(),
-                        new NettyToAzureCoreHttpHeadersWrapper(reactorNettyResponse.responseHeaders()))
+                return reactorNettyConnection.inbound().receive().aggregate().asByteArray()
                     .doFinally(ignored -> closeConnection(reactorNettyConnection))
+                    .switchIfEmpty(Mono.fromSupplier(() -> new byte[0]))
```

Comment:
Let's use a constant for the empty byte array and then we can switch to `Mono.just()`.